### DISCUSSION TITLE
_PRINT_START_PHASE_INIT crashes Klipper / Moonraker

### DISCUSSION THREAD

**Discussion by oldschool1964** [jschuh/klipper-macros#210] [General]
I'm trying to get the _PRINT_START_PHASE_INIT approach working. I can print OK without this macro, so I know the problem is inside of this or the other steps of the macro.

When I issue the command, the printer heats to hotend 115 and bed 60 degrees celcius. I'm not sure where the 115 number comes from, but that's not my worry at this point. Once it heats up, the Klipper immediately shuts down and requires a firmware_restart. I've looked at the logs, and Klipper logs don't show anything. The only clue I get is from the Moonraker plugin. Here's the relevant lines:

`2023-12-14 16:52:57,880 [app.py:post()] - Processing Uploaded File: /Untitled.ufp
2023-12-14 16:52:58,216 [shell_command.py:pipe_data_received()] - INFO:metadata:Object Processing is disabled

2023-12-14 16:52:58,404 [shell_command.py:_check_proc_success()] - Command (/home/pi/moonraker-env/bin/python /home/pi/moonraker/moonraker/components/file_manager/metadata.py -p /home/pi/printer_data/gcodes -f "Untitled.gcode" -u "/tmp/moonraker.upload-1205.mru") successfully finished
2023-12-14 16:52:58,431 [app.py:log_request()] - 201 POST /server/files/upload (192.168.1.178) [_TRUSTED_USER_] 615.51ms
2023-12-14 16:52:58,439 [common.py:build_error()] - JSON-RPC Request Error - Requested Method: server.files.metadata, Code: -32601, Message: Metadata not available for <.thumbs/Untitled.png>
2023-12-14 16:52:58,466 [job_state.py:_status_update()] - Job Started: Untitled.gcode
2023-12-14 16:56:17,014 [klippy_connection.py:_process_status_update()] - Klippy has shutdown
2023-12-14 16:56:17,029 [websockets.py:on_close()] - Websocket Closed: ID: 3886795240 Close Code: 1000, Close Reason: None, Pong Time Elapsed: 2.82
2023-12-14 16:56:17,032 [proc_stats.py:_handle_shutdown()] -`

I'm using Cura with the Moonraker plugin to send my slicer file to Klipper. Any idea what's going on? I can provide exhaustive logs if required. It looks like there's a problem with the metadata for the image that is sent with the thumbnail of the print maybe? I'm not sure what's going on, I just know that I get a hard crash after my printer heats up during the first phase of the init routine.

**Comment by oldschool1964** [comment]
[klippy (1).log](https://github.com/jschuh/klipper-macros/files/13679113/klippy.1.log)
[moonraker (4).log](https://github.com/jschuh/klipper-macros/files/13679115/moonraker.4.log)
Logs attached.

**Comment by jschuh** [comment]
You're getting `MCU 'mcu' shutdown: Rescheduled timer in the past` errors, which means your MCU can't keep up with your host (or vice versa). That doesn't really have anything to do with these macros, except that when an MCU or host is struggling it can be pushed over the edge by any little thing.

FWIW, the macros themselves should be adding almost no overhead to your system. As I case in point, they run just fine on my weakest provisioned machines, which have 16-bit MCUs and first-gen OPI Zero (512MB) hosts.

**Reply by oldschool1964** [reply]
I'm running on a RPi4, so it shouldn't be choking on anything. I removed all of the LED commands that might have caused my Pico to go overboard, and I'm still having problems. I just tested right now using Cura, and here's my machine settings:

```
LIGHTS_OFF
LIGHT_HIGH

_PRINT_START_PHASE_INIT EXTRUDER={material_print_temperature_layer_0} BED={material_bed_temperature_layer_0} NOZZLE_SIZE={machine_nozzle_size}
; Insert custom gcode here.
;NEOPIXEL_DISPLAY LED="overhead" TYPE=extruder_temp MODE=glow

_PRINT_START_PHASE_PREHEAT
; Insert custom gcode here.
;LIGHTS_OFF
;LIGHT_HIGH

_PRINT_START_PHASE_PROBING
; Insert custom gcode here.

_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.

_PRINT_START_PHASE_PURGE

; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically calculate and 
; perform the purge (e.g. if using a Mosaic Palette, which requires the slicer
; to generate the purge).

M117 Printing...

NEOPIXEL_DISPLAY LED="overhead" TYPE=print_percent MODE=progress
```
I get the error:
```
MCU 'mcu' shutdown: Rescheduled timer in the past
This generally occurs when the micro-controller has been
requested to step at a rate higher than it is capable of
obtaining.
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Printer is shutdown
```
Troubleshooting the commands, I can get through the preheat (which heats the printer to 115/62, and then 115/60). When we get to the PROBE command, the temp jumps to 260C and then immediately shuts down, so the _PRINT_START_PHASE_PROBING command is causing my crash.

Thank you for your assistance, and I would appreciate your assistance with this new information. Thank you!
